### ISSUE TITLE
foreman_host does not power on vmware based hosts before attaching bootdisks

### ISSUE THREAD

**Issue by childofthewired** [theforeman/foreman-ansible-modules#818]
##### SUMMARY
When using theforeman.foreman.foreman_host with Satellite 6.6.3 and provision_method "bootdisk"

The module will upload the bootdisk to the datastore and reconfigure the VM to boot from the bootdisk before powering the VM on. This is out of sequence from how formam/satellite typically does this, and as such, virtual machines will not boot and provision.

Typically the build goes like this:
Foreman builds the bootdisk and the vm
foreman uploads the boot disk to the datastore
foreman powers on the VM 
foreman reconfigures the VM to boot from the bootdisk ISO

##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION
Ansible 3.6.4

##### KATELLO/FOREMAN VERSION
tfm-rubygem-katello-3.12.0.30-1.el7sat.noarch

##### APYPIE VERSION```

Version: 0.2.2
##### STEPS TO REPRODUCE

```
- name: Playbook that Creates hosts in Satellite using Ansible Foreman Modules

  hosts: localhost

  vars:

    sat_username: "<id>"

    sat_password: "<pwd>"

    sat_url: satelliteFQDN

    host_fqdn: hostFQDN

 

  tasks:

  - name: Create a host in Satellite6

    theforeman.foreman.foreman_host:

      username: "{{ sat_username }}"

      password: "{{ sat_password }}"

      server_url: "{{ sat_url }}"

      name: "{{ host_fqdn }}"

      ip: "192.168.109.7"

      provision_method: "bootdisk"

      hostgroup: "Development_Servers/RHEL_7/Infrastructure/VRA_testing"

      organization: "Geisinger"

      location: "DataCenter"

      state: present

      validate_certs: no

      build: true

 

  - name: Power on the built host

    theforeman.foreman.foreman_host_power:

      username: "{{ sat_username }}"

      password: "{{ sat_password }}"

      server_url: "{{ sat_url }}"

      hostname: "{{ host_fqdn }}"

      state: on

      validate_certs: no

 

```

Actual results:

See above.

**Comment by dmaple-gci** [comment]
I have the same issue with Ansible 2.9, Satellite 6.6.2, katello 3.12.0.37.

**Comment by evgeni** [comment]
@Fobhep @hlawatschek you guys use VMWare, right?

**Comment by Fobhep** [comment]
@evgeni yes we do. I will have a look at this.

**Comment by FlatKey** [comment]
I have a similar issue, but the bootdisk will never connected even after using foreman_host_power to start the VM.

Ansible 2.9.0
Apypie 0.2.2
Foreman 2.1.0
VMware vSphere 6.5.0

**Comment by Fobhep** [comment]
Hi all,

sorry that took a while. But I reached out to some colleagues of mine that actually are using bootdisk and I have a possible solution.
There are some `vmware` only features for certain compute attributes.
https://access.redhat.com/documentation/en-us/red_hat_satellite/6.2/html/provisioning_guide/cli_params

eg
```
start | Boolean value that defines whether to start the machine or not
```

Could you maybe try and add that to the compute_attributes that are associated with your hostgroup.

```
- name: "Create compute attribute"
  compute_attribute:
    username: "admin"
    password: "changeme"
    server_url: "https://foreman.example.com"
    compute_profile: "Test Compute Profile"
    compute_resource: "Test Compute Resource"
    vm_attrs:
      start: 'true' 
    state: present
```

Maybe that could ease the pain for now.
However eventually the host module should also be able to add compute attributes on the fly.

**Comment by childofthewired** [comment]
Running the equivalent hammer command requires the following to be appended to start the virtual machine and attach the disk in the proper order.

`--compute-attributes "start=1"
`

**Comment by Fobhep** [comment]
that is exactly what I meant with the link I posted [above](access.redhat.com/documentation/en-us/red_hat_satellite/6.2/html/provisioning_guide/cli_params) mentioning the vmware specific attributes.
Did you already try including that in your compute attributes?

**Comment by childofthewired** [comment]
The suggested playbook does not work:

```

PLAY [Playbook that Creates hosts in Satellite using Ansible Foreman Modules] ******************************************************

 

TASK [Gathering Facts] *************************************************************************************************************

ok: [localhost]

 

TASK [Create compute attribute] ****************************************************************************************************

fatal: [localhost]: FAILED! => {"changed": false, "error": {"message": "Fog::Vsphere::Compute::NotFound"}, "msg": "Failed to show 
r}


PLAY RECAP *************************************************************************************************************************

localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0`


```


Checking hammer for the equivalent commands: 

When I create a host - 

`
hammer host create --name test --compute-profilfe xyz --compute-resource vmware-host --compute-attributes  --compute-attributes "start=1"
`
Works {excluding my hostgroup info} and starts the vmware VM for provisioning.

the help pages for 'hammer host create' show that the '--compute-attributes "start=1" is available.

When we look at the help pages for 'hammer compute-profile values create'
 --compute-attributes "start=1" is not an option for VMware.

This appears to need to be done at host creation.

**Comment by evgeni** [comment]
With https://github.com/theforeman/foreman-ansible-modules/pull/932 merged, could you please try adding the following to the `host` module call and see if it changes anything:

```
    compute_attributes:
       start: "1"
```

**Comment by gardar** [comment]
> With #932 merged, could you please try adding the following to the `host` module call and see if it changes anything:
> 
> ```
>     compute_attributes:
>        start: "1"
> ```

I'm not OP but I came here with the same issue and I can confirm that this works for me, thanks!

**Comment by evgeni** [comment]
Based on feedback, I am going to close this as fixed by #932, thanks!
If the issue still happens to you, please open a new issue and reference this one for background.
